crates/uv-distribution/src/distribution_database.rs

```diff
@@ -320,6 +323,8 @@ impl<'a, Context: BuildContext + Send + Sync> DistributionDatabase<'a, Context>
         dist: &Dist,
     ) -> Result<(Metadata23, Option<Url>), Error> {
         match dist {
+            // TODO(zanieb): `dist.metadata()` should return us the right error kind so we can use `Error::Metadata` here
+            Dist::Installed(dist) => Ok(dist.metadata().map(|metadata| (metadata, None)).unwrap()),
```

Comment:
E.g., I think this should probably happen at a level above? Passing an installed distribution to this remote fetcher feels off.